manual: develop/firmware: add note to fedora tab about path

purdeaandrei opened the pull request:
For reference:
Fedora changelog:
![image](https://github.com/user-attachments/assets/ae4a8d03-be88-4a85-9322-456280bd712f)
Fedora readme for SDCC:
![image](https://github.com/user-attachments/assets/e7ed1571-7140-4ce9-a78a-929a9ef4082b)

purdeaandrei commented:
hmm, I hope the nested tap/note/code works correctly

purdeaandrei commented:
When switching to the fedora tab:

![image](https://github.com/user-attachments/assets/bf771b47-0711-4009-a149-184a8300f76f)

purdeaandrei commented:
![image](https://github.com/user-attachments/assets/3571596f-56fa-4ad3-bb6e-b89e627e58b5)

whitequark commented on docs/manual/src/develop/firmware.rst:
```diff
@@ -75,6 +75,14 @@ The firmware can only be built on a Unix-like system; to develop the firmware on
 
+    .. note ::
+
+        On Fedora the sdcc package installs its binaries into ``/usr/libexec/sdcc``, and it installs shellscripts into ``/usr/bin`` with the ``sdcc-`` prefix, that pass execution onto the real binaries in libexec. The firmware build system relies on sdcc binaries having the non-prefixed names, so it's recommended to append ``/usr/libexec/sdcc`` to the ``PATH`` environment variable. If you use bas ...
```
This is a lot of text that's not really necessary. The additional explanation adds very little, and anyone building the firmware themselves is going to know how to manipulate `PATH`.

```suggestion
        On Fedora, the `sdcc` package installs the tools into a different location, which must be added to the search path:
```

whitequark commented on docs/manual/src/develop/firmware.rst:
```diff
@@ -75,6 +75,14 @@ The firmware can only be built on a Unix-like system; to develop the firmware on
+
+        On Fedora the sdcc package installs its binaries into ``/usr/libexec/sdcc``, and it installs shellscripts into ``/usr/bin`` with the ``sdcc-`` prefix, that pass execution onto the real binaries in libexec. The firmware build system relies on sdcc binaries having the non-prefixed names, so it's recommended to append ``/usr/libexec/sdcc`` to the ``PATH`` environment variable. If you use bas ...
+
+        .. code:: bash
```
```suggestion
        .. code:: console
```

purdeaandrei commented:
Added your suggestions, and squashed.

It now looks like this: 
![image](https://github.com/user-attachments/assets/59c863c9-22b4-43b5-8c8e-ab172c82f1de)

whitequark commented:
Thanks!
